src/plugin.ts

```diff
@@ -82,14 +84,18 @@ class LiftPlugin {
             "lift:eject:eject": this.eject.bind(this),
         };
 
-        // TODO variables should be resolved just before deploying each provider
-        // else we might get outdated values
         this.configurationVariablesSources = {
-            // TODO these 2 variable sources should be merged eventually
             construct: {
                 resolve: this.resolveReference.bind(this),
             },
         };
+        this.variableResolvers = {
+            construct: (fullVariable) => {
+                const address = fullVariable.split(":")[1];
+
+                return Promise.resolve(this.resolveReference({ address }).value);
```

Comment:
does this need to be async ? 
```suggestion
                return this.resolveReference({ address }).value;
```

Reply:
Yes it does, at least that's how other variable resolvers seem to be implemented, and things didn't work when I did not return a promise.